c/ovm.c

```diff
@@ -857,6 +860,8 @@ static word prim_sys(int op, word a, word b, word c) {
          fd1 = c != IFALSE ? dup2(fd0, fd1) : fcntl(fd0, F_DUPFD, fd1);
          if (fd1 == -1)
             return IFALSE;
+         if (fd1 < 3)
+            toggle_blocking(fd1, 1);
```

Comment:
The ugly thing with this is, that it leaves `fd0` in blocking mode, too. But since the usual flow is to close the old port after `dup()`, it is still better then having non-blocking stdio – especially in the case of `exec()`.